bbq/compiler/desugar.go

```diff
@@ -131,22 +175,72 @@ func (d *Desugar) VisitVariableDeclaration(declaration *ast.VariableDeclaration)
 func (d *Desugar) VisitFunctionDeclaration(declaration *ast.FunctionDeclaration, _ bool) ast.Declaration {
 	funcBlock := declaration.FunctionBlock
 	funcName := declaration.Identifier.Identifier
+	parameterList := declaration.ParameterList
+	returnTypeAnnotation := declaration.ReturnTypeAnnotation
 
-	preConditions := d.desugarPreConditions(
+	functionType := d.elaboration.FunctionDeclarationFunctionType(declaration)
+
+	// If this is an interface-method without a body,
+	// then do not generate a function for it.
+	if d.enclosingInterfaceType != nil && !funcBlock.HasStatements() {
+		return nil
+	}
+
+	modifiedFuncBlock := d.desugarFunctionBlock(
```

Comment:
The diff here looks complex/more than what it actually is. The actual change here was just to move the function body desugaring logic to `desugarFunctionBlock` function, so that it can be re-used for functio expressions as well.